### PR TITLE
Fix BH tests

### PR DESCRIPTION
**Pull request by broskoTT**
We still don't have CI for BH, so we easily forget to test if all tests (including new API tests) still work for Blackhole.
I was testing something unrelated on a blackhole machine and found these disrepancies.
Related to #154 and #229

### CODE REVIEW

File: tests/api/test_chip.cpp [hunk 1]
```diff
@@ -44,12 +44,16 @@ inline std::unique_ptr<tt_ClusterDescriptor> get_cluster_desc() {
...
-    // TODO: remove getting manually cluster descriptor from yaml.
-    std::string yaml_path = test_utils::GetClusterDescYAML();
+    // TODO: Remove different branch for different archs
     std::unique_ptr<tt_ClusterDescriptor> cluster_desc;
     if (device_arch == tt::ARCH::GRAYSKULL) {
         cluster_desc = tt_ClusterDescriptor::create_for_grayskull_cluster(pci_device_ids_set, pci_device_ids);
+    } else if (device_arch == tt::ARCH::BLACKHOLE) {
+        std::string yaml_path = test_utils::GetAbsPath("blackhole_1chip_cluster.yaml");
+        cluster_desc = tt_ClusterDescriptor::create_from_yaml(yaml_path);
     } else {
```
Comment (pjanevskiTT) [inline tests/api/test_chip.cpp:54]: would explicit 

```cpp
if (device_arch == tt::ARCH::WORMHOLE_B0)
```

be more readable here? Same for all other changes

**Review by pjanevskiTT — APPROVED** [review]

File: tests/blackhole/test_silicon_driver_bh.cpp [hunk 2]
```diff
@@ -229,10 +229,9 @@ TEST(SiliconDriverBH, UnalignedStaticTLB_RW) {
                 // Statically mapping a 1MB TLB to this core, starting from address NCRISC_FIRMWARE_BASE.  
                 device.configure_tlb(i, core, get_static_tlb_index_callback(core), l1_mem::address_map::NCRISC_FIRMWARE_BASE);
             }
+            device.setup_core_to_tlb_map(i, get_static_tlb_index_callback);
         } 
     }
-
-    device.setup_core_to_tlb_map(get_static_tlb_index_callback);
```
Comment (joelsmithTT) [inline tests/blackhole/test_silicon_driver_bh.cpp:235]: How did this compile?

**Review by joelsmithTT — APPROVED** [review]
does create-ethernet-map not work for BH?

File: tests/blackhole/test_silicon_driver_bh.cpp (hunk 2 above)
Comment (broskoTT) [reply tests/blackhole/test_silicon_driver_bh.cpp:235]: It didn't compile before, we don't check for blackhole :) Actually I can add a build for blackhole, even though we don't have blackhole cards to run on.

File: tests/api/test_chip.cpp (hunk 1 above)
Comment (broskoTT) [reply tests/api/test_chip.cpp:54]: Well actually not. The else branch which is following is not "something specific for wormhole", rather it is a "standard" way of doing things. Grayskull and Blackhole have something specific done to them. 
To be specific, in one of the following PRs (specifically when cem starts working for BH) I'd expect to remove the blackhole branch. Maybe in some of the following changes to remove Grayskull branch as well.

**Pull request merged**